assignments/this.js

```diff
@@ -13,14 +13,61 @@
 
 // code example for Window Binding
 
+function action(param) {
+    "use strick";
+    console.log(this);
+    return `Create the ${param}`;
+}
+
+console.log(action("action"));
+
 // Principle 2
 
 // code example for Implicit Binding
 
+const orc = {
+    name: "Kil",
+    age: "38",
+    attack: function() {
+        return `${this.name} is going to attack because he is ${this.age}`;
+    }
+}
+
+console.log(orc.attack());
+
 // Principle 3
 
 // code example for New Binding
 
+const student = {
+    name: "Sean"
+}
+
+const worker = {
+    name: "Bob"
+}
+
+const abilities = ["Doin this", "doin that", "and da otha ting"];
+
+function activate(abilities1, abilities2, abilities3) {
+    return `Hi! My name is: ${this.name} and this is what I do: ${abilities1}, ${abilities2}, ${abilities3},`
+}
+
+console.log(activate.call(student, abilities))
```

Comment:
adding ... to abilities here will make your console log perfect! `console.log(activate.call(student, ...abilities))` ... is a spread operator and basically "opens" up what you apply it to. this way we pass the student to call and 3 arguments (3 abilities) instead of student and one argument of all 3 abilities at once